cmd/posix-errors.go

```diff
@@ -47,8 +45,7 @@ func isSysErrOpNotSupported(err error) bool {
 // No space left on device error
 func isSysErrNoSpace(err error) bool {
 	if pathErr, ok := err.(*os.PathError); ok {
-		switch pathErr.Err {
-		case syscall.ENOSPC:
+		if pathErr.Err == syscall.ENOSPC {
```

Comment:
Further simplification is possible simply do `return pathErr.Err == syscall.ENOSPC`